lib/search/index-class.js

```diff
@@ -0,0 +1,140 @@
+/*!
+ * Copyright 2015 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*!
+ * @module search/index
+ */
+
+'use strict';
+
+var extend = require('extend');
+
+/**
+ * @type {module:search/document}
+ * @private
+ */
+var Document = require('./document.js');
+
+/**
+ * @type {module:common/util}
+ * @private
+ */
+var util = require('../common/util.js');
+
+function Index(search, id) {
+  this.search_ = search;
+  this.id = id;
+}
+
+Index.prototype.createDocument = function(document, callback) {
```

Comment:
Only if we do a browser build. In which case, there are a lot of variables there; we might be able to rename it at the time of export (to document_ or something). It's something we can figure out quite easily when that time comes.